contracts/modules/cover/CoverOrder.sol

```diff
@@ -0,0 +1,225 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.18;
+
+import "@openzeppelin/contracts-v4/security/ReentrancyGuard.sol";
+import "@openzeppelin/contracts-v4/utils/cryptography/ECDSA.sol";
+import "@openzeppelin/contracts-v4/token/ERC20/IERC20.sol";
+import "@openzeppelin/contracts-v4/token/ERC20/utils/SafeERC20.sol";
+import "@openzeppelin/contracts-v4/access/Ownable.sol";
+
+import "../../interfaces/ICoverOrder.sol";
+import "../../interfaces/IMemberRoles.sol";
+import "../../interfaces/INXMMaster.sol";
+import "../../interfaces/INXMToken.sol";
+import "../../interfaces/IWeth.sol";
+import "../../interfaces/IPool.sol";
+
+contract CoverOrder is ICoverOrder, Ownable {
+  using ECDSA for bytes32;
+  using SafeERC20 for IERC20;
+
+  /* ========== STATE VARIABLES ========== */
+  mapping(bytes32 => bool) public executedOrders;
+
+  /* ========== IMMUTABLES ========== */
+  ICover public immutable cover;
+  IMemberRoles public immutable memberRoles;
+  INXMToken public immutable nxmToken;
+  INXMMaster public immutable master;
+  IWeth public immutable weth;
+
+  /* ========== CONSTANTS ========== */
+  address public constant ETH = 0xEeeeeEeeeEeEeeEeEeEeeEEEeeeeEeeeeeeeEEeE;
+  uint private constant ETH_ASSET_ID = 0;
+  uint private constant NXM_ASSET_ID = type(uint8).max;
+
+  /* ========== CONSTRUCTOR ========== */
+  constructor(
+    address _cover,
+    address _memberRoles,
+    address _nxmToken,
+    address _master,
+    address _weth,
+    address _owner
+  ) {
+    master = INXMMaster(_master);
+    cover = ICover(_cover);
+    memberRoles = IMemberRoles(_memberRoles);
+    nxmToken = INXMToken(_nxmToken);
+    weth = IWeth(_weth);
+    transferOwnership(_owner);
+  }
+
+  /// @notice Verifies and executes the order to buy cover on behalf of the creator of limit order
+  /// @param params Cover buy parameters
+  /// @param poolAllocationRequests Pool allocations for the cover
+  /// @param limitOrderId The ID of the limit order
+  /// @param signature The signature of the order
+  function executeOrder(
+    BuyCoverParams calldata params,
+    PoolAllocationRequest[] calldata poolAllocationRequests,
+    uint limitOrderId,
+    bytes calldata signature
+  ) external payable returns (uint coverId) {
+
+    if (params.owner == address(0) || params.owner == address(this)) {
+      revert InvalidOwnerAddress();
+    }
+
+    if (params.paymentAsset == NXM_ASSET_ID) {
+      revert InvalidPaymentAsset();
+    }
```

Comment:
can you remind me please why we don't want to use nxm for payment here please?

Reply:
I had a note for this, but double checked it's not a requirement so I removed it